mlx/traceability.py

```diff
@@ -76,6 +76,10 @@ class ItemTree(nodes.General, nodes.Element):
     '''Tree-view on documentation items'''
     pass
 
+class ItemLink(nodes.General, nodes.Element):
```

Comment:
not related to this line, but more conceptual:

note to self: are we sure all items exist (source and target) when the item-link starts adding relations to the database...

Reply:
The ``add_relation`` function performs checks on the source and relationship type. For some reason it doesn't do it for the target. 

I assume this is intentional. Any idea why?

Reply:
When items are defined in RST, we parse them in the plugin one by one. While parsing we update the relations of this item. So we store where the items link to = forward relation.  But we also need to store the reverse relation. So to the target item(s) we add the 'reverse' relation to the source item.

Depending on the order of parsing the RST files and items, a target item might not exist in the 'database' yet. So the script will add a placeholder item and add the 'reverse' relation to it. Later, when parsing more items the placeholder item is updated with the actual content.

A self-check performs check on the database to make sure there is no placeholder items left.

For the above reason we cannot assert that the target item is in the database so we don't check it. The source item and relation kind should exist, that is what we can assert.

This information is - or should be - in the readme somehow. At least a mention of placeholder items...